package/MDAnalysis/lib/qcprot.pyx

```diff
@@ -186,15 +192,15 @@ def InnerProduct(np.ndarray[np.float64_t,ndim=1] A,
 
     if (weight is not None):
         for i in range(N):
-            x1 = weight[i] * coords1[0,i]
-            y1 = weight[i] * coords1[1,i]
-            z1 = weight[i] * coords1[2,i]
+            x1 = weight[i] * coords1[i,0]
```

Comment:
I haven't checked, probably C-style. But the changes speed up the code. Do you think that there's room for more speed-ups?

By the way, is it a good idea to use numpy array operations in cython or should this be done explicitly on individual array elements using loops as in the code below?